### PR TITLE
Fix stylelint

### PR DESCRIPTION
**Pull request by Gudahtt**
* Stylelint: Ignore only top-level directories
  The `.stylelintignore` entries lacked leading slashes, so most of the UI code was ignored (because it fell under the `ui/app` directory, and `app/` was ignored.

  The leading slashes ensure only the intended top-level directories are ignored.

* Simplify stylelint rules
  We use the `stylelint-config-standard` rule-set, so most commonly-used stylelint rules are inherited from that.

  Some of the removed rules were redundant, some of them were more strict than the rules in `standard` and we hadn't been following them in practice, and some were obsolete.

* Convert stylelint config to JavaScript
  JavaScript is a bit easier than JSON to work with, as it allows comments.

  This was also done to make it easier to merge in the `stylelint-config-standard`, which is also in JavaScript.

* Inline `stylelint-config-standard`
  I intend to go through each of these rules one-by-one, which is easier with all of these rules inlined. Selectively overriding/disabling them would have been messy.

* Comment out rules that aren't current working
  These rules have been temporarily disabled. They will be re-renabled one-by-one as they are fixed. This was done to make it easier to split these changes among separate PRs, as many of the rules require extensive functional changes.

### CODE REVIEW

**Comment by Gudahtt** [comment]
Lots of lint errors to fix here still

**Comment by whymarrh** [comment]
Huh, this makes a lot of sense

File: .stylelintignore [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-fonts/
-images/
-node_modules/
-notices/
-test/
+/app
+/development
+/dist
+/docs
+/fonts
```
Comment (Gudahtt) [inline .stylelintignore]: Ignore this - it is a test
```suggestion
/dist/
/docs/
/fonts/
```

**Review by brad-decker — APPROVED** [review]

**Comment by whymarrh** [comment]
Is the plan to incrementally re-enable some of the rules in separate PRs?

**Comment by Gudahtt** [comment]
Yes, exactly. I'll probably do one batch PR of anything that I can auto-fix without functional changes, then do the rest one-by-one.

**Review by whymarrh — APPROVED** [review]

**Pull request merged**